include/boost/url/detail/impl/pct_encoding.ipp

```diff
@@ -81,6 +82,224 @@ key_equal_encoded(
     }
 }
 
+int
+pct_decode_compare_unchecked(
```

Comment:
these really don't belong here in pct_encoding.ipp, anything percent-encoding related that is strictly for comparison, normalization, or hash functions should have its own header, for example `detail/normalize.hpp`. Or something.